flink-end-to-end-tests/test-scripts/test_tpcds.sh

```diff
@@ -24,86 +24,153 @@ USE_TABLE_STATS=true
+
+    local scheduler="$1"
+
+    set_config_key "jobmanager.scheduler" "${scheduler}"
+    set_config_key "taskmanager.memory.process.size" "4096m"
+    set_config_key "taskmanager.memory.network.fraction" "0.2"
+
+    if [ "${scheduler}" == "Default" ]; then
+        set_config_key "taskmanager.numberOfTaskSlots" "4"
+        set_config_key "parallelism.default" "4"
+    elif [ "${scheduler}" == "AdaptiveBatch" ]; then
+        set_config_key "taskmanager.numberOfTaskSlots" "8"
+        set_config_key "parallelism.default" "-1"
+        set_config_key "jobmanager.adaptive-batch-scheduler.max-parallelism" "8"
+        set_config_key "jobmanager.adaptive-batch-scheduler.avg-data-volume-per-task" "6m"
+        set_config_key "jobmanager.adaptive-batch-scheduler.speculative.enabled" "true"
+        set_config_key "jobmanager.adaptive-batch-scheduler.speculative.block-slow-node-duration" "0s"
+        set_config_key "slow-task-detector.execution-time.baseline-ratio" "0.0"
+        set_config_key "slow-task-detector.execution-time.baseline-lower-bound" "0s"
+    else
+        echo "ERROR: Scheduler ${scheduler} is unsupported for tpcds test. Aborting..."
+        exit 1
+    fi
+
+    start_cluster
+
+
+    ################################################################################
+    # Run TPC-DS SQL
+    ################################################################################
+
+    echo "[INFO] Runing TPC-DS queries..."
+
+    RESULT_DIR="$TARGET_DIR/result"
+    mkdir -p "$RESULT_DIR"
+
+    $FLINK_DIR/bin/flink run -c org.apache.flink.table.tpcds.TpcdsTestProgram "$TARGET_DIR/TpcdsTestProgram.jar" -sourceTablePath "$TPCDS_DATA_DIR" -queryPath "$TPCDS_QUERY_DIR" -sinkTablePath "$RESULT_DIR" -useTableStats "$USE_TABLE_STATS"
+
+    ################################################################################
+    # validate result
+    ################################################################################
+    QUALIFIED_ANSWER_DIR="$TARGET_DIR/answer_set_qualified"
+    mkdir -p "$QUALIFIED_ANSWER_DIR"
+
+    java -cp "$TARGET_DIR/TpcdsTestProgram.jar:$TARGET_DIR/lib/*" org.apache.flink.table.tpcds.utils.AnswerFormatter -originDir "$ORGIN_ANSWER_DIR" -destDir "$QUALIFIED_ANSWER_DIR"
+
+    java -cp "$TARGET_DIR/TpcdsTestProgram.jar:$TARGET_DIR/lib/*" org.apache.flink.table.tpcds.utils.TpcdsResultComparator -expectedDir "$QUALIFIED_ANSWER_DIR" -actualDir "$RESULT_DIR"
+
+    ################################################################################
+    # Clean-up generated data folder
+    ################################################################################
+
+    rm -rf "${TPCDS_DATA_DIR}"
+    echo "Deleted all files under $TPCDS_DATA_DIR"
+}
+
+function check_logs_for_exceptions_for_adaptive_batch_scheduler {
+    echo "Checking for exceptions for adaptive batch scheduler..."
+    exception_count=$(grep -rv "GroupCoordinatorNotAvailableException" $FLINK_LOG_DIR \
```

Comment:
It is possible to re-use the whitelisted exceptions in `check_logs_for_exceptions`, in `common.sh`?